### PR TITLE
add support for Cloud Armor Edge Policies

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/10761

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added support for field `type` to `google_compute_security_policy`
```
```release-note:enhancement
compute: added support for attaching a `edge_security_policy` to `google_compute_backend_bucket`
```

### CODE REVIEW

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks like tests are failing?

**Comment by ScottSuarez** [comment]
Everything should be kosher here @slevenick ^-^

File: mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// security_policy isn't set by Create / Update
```
Comment (slevenick) [inline mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb:1]: This exists in the post_create, but what if the user goes from this field being unset on a bucket to setting it? Should the field be ForceNew or do we also need this in the Update call?

File: mmv1/third_party/terraform/tests/resource_compute_security_policy_test.go.erb [hunk 2]
```diff
@@ -197,6 +197,7 @@ func testAccComputeSecurityPolicy_basic(spName string) string {
 resource "google_compute_security_policy" "policy" {
   name        = "%s"
   description = "basic security policy"
+	type        = "CLOUD_ARMOR_EDGE"
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_compute_security_policy_test.go.erb:200]: Spacing is a little weird here

File: mmv1/third_party/terraform/tests/resource_compute_backend_bucket_test.go [hunk 3]
```diff
@@ -201,3 +232,27 @@ resource "google_storage_bucket" "bucket" {
...
+resource "google_storage_bucket" "image_bucket" {
+  name     = "%s"
+  location = "EU"
+}
+
+
+resource "google_compute_security_policy" "policy" {
+  name        = "%s"
+  description = "basic security policy"
+	type = "CLOUD_ARMOR_EDGE"
```
Comment (slevenick) [inline mmv1/third_party/terraform/tests/resource_compute_backend_bucket_test.go:255]: Spacing here as well

File: mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb (hunk 1 above)
Comment (slevenick) [reply mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb:1]: Oh this is used as post_update as well, makes sense

File: mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+if o, n := d.GetChange("edge_security_policy"); o.(string) != n.(string) {
+		pol, err := ParseSecurityPolicyFieldValue(n.(string), d, config)
+  if err != nil {
+    return errwrap.Wrapf("Error parsing Backend Service security policy: {{err}}", err)
+  }
+
+  spr := emptySecurityPolicyReference()
+  spr.SecurityPolicy = pol.RelativeLink()
+  op, err := config.NewComputeClient(userAgent).BackendBuckets.SetEdgeSecurityPolicy(project, obj["name"].(string), spr).Do()
```
Comment (slevenick) [inline mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb:10]: Can we do this through the HTTP client directly? It feels a little awkward to add a dependency on the client libraries to a mmv1-based resource

File: mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb (hunk 1 above)
Comment (slevenick) [reply mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb:1]: Can the update part of this be done by setting `update_url` on this specific field in the terraform.yaml? It allows calling a specific endpoint for a specific field

File: mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb (hunk 4 above)
Comment (ScottSuarez) [reply mmv1/templates/terraform/post_create/compute_backend_bucket_security_policy.go.erb:10]: This is already a pattern in another resource https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/post_create/compute_backend_service_security_policy.go.erb#L8-L9

There is where I adapted the code from

**Comment by ScottSuarez** [comment]
fixed spacing, awaiting response for other suggestions

**Pull request merged**